internal/mode/static/telemetry/collector.go

```diff
@@ -0,0 +1,148 @@
+package telemetry
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	v1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/nginxinc/nginx-gateway-fabric/internal/mode/static/state/dataplane"
+	"github.com/nginxinc/nginx-gateway-fabric/internal/mode/static/state/graph"
+)
+
+//go:generate go run github.com/maxbrunsfeld/counterfeiter/v6 . GraphGetter
+
+// GraphGetter gets the latest Graph.
+type GraphGetter interface {
+	GetLatestGraph() *graph.Graph
+}
+
+//go:generate go run github.com/maxbrunsfeld/counterfeiter/v6 . ConfigurationGetter
+
+// ConfigurationGetter gets the latest Configuration.
+type ConfigurationGetter interface {
+	GetLatestConfiguration() *dataplane.Configuration
+}
+
+// NGFResourceCounts stores the counts of all relevant resources that NGF processes and generates configuration from.
+type NGFResourceCounts struct {
+	Gateways       int
+	GatewayClasses int
+	HTTPRoutes     int
+	Secrets        int
+	Services       int
+	Endpoints      int
```

Comment:
Let's add a comment here, because Endpoints is ambiguous - is Endpoints a resource or IP:port?
// Endpoints include the total count of Endpoints (IP:port) across all referenced Services.